nilearn/plotting/tests/test_html_surface.py

```diff
@@ -0,0 +1,377 @@
+        html.save_as_html(tmpfile)
+        with open(tmpfile) as f:
+            saved = f.read()
+        assert saved == html.get_standalone()
+    finally:
+        os.remove(tmpfile)
+    assert "INSERT" not in html.html
+    assert html.get_standalone() == html.html
+    assert html._repr_html_() == html.get_iframe()
+    assert str(html) == html.get_standalone()
+    assert '<meta charset="UTF-8" />' in str(html)
+    _check_open_in_browser(html)
+    resized = html.resize(3, 17)
+    assert resized is html
+    assert (html.width, html.height) == (3, 17)
+    assert "width=3 height=17" in html.get_iframe()
+    assert "width=33 height=37" in html.get_iframe(33, 37)
+    if not LXML_INSTALLED:
+        return
+    root = etree.HTML(html.html)
+    head = root.find('head')
+    assert len(head.findall('script')) == 5
+    body = root.find('body')
+    div = body.find('div')
+    assert ('id', 'surface-plot') in div.items()
+    selects = body.findall('select')
+    assert len(selects) == 3
+    hemi = selects[0]
+    assert ('id', 'select-hemisphere') in hemi.items()
+    assert len(hemi.findall('option')) == 2
+    kind = selects[1]
+    assert ('id', 'select-kind') in kind.items()
+    assert len(kind.findall('option')) == 2
+    view = selects[2]
+    assert ('id', 'select-view') in view.items()
+    assert len(view.findall('option')) == 7
+
+
+def _open_mock(f):
+    print('opened {}'.format(f))
+
+
+def _check_open_in_browser(html):
+    wb_open = webbrowser.open
+    webbrowser.open = _open_mock
+    try:
+        html.open_in_browser(temp_file_lifetime=None)
+        temp_file = html._temp_file
+        assert html._temp_file is not None
+        assert os.path.isfile(temp_file)
+        html.remove_temp_file()
+        assert html._temp_file is None
+        assert not os.path.isfile(temp_file)
+        html.remove_temp_file()
+        html._temp_file = 'aaaaaaaaaaaaaaaaaaaaaa'
+        html.remove_temp_file()
+    finally:
+        webbrowser.open = wb_open
+        try:
+            os.remove(temp_file)
+        except Exception:
+            pass
+
+
+def test_temp_file_removing():
+    html = html_surface.SurfaceView('hello')
+    wb_open = webbrowser.open
+    webbrowser.open = _open_mock
+    try:
+        html.open_in_browser(temp_file_lifetime=1.5)
+        assert os.path.isfile(html._temp_file)
+        time.sleep(1.6)
+        assert not os.path.isfile(html._temp_file)
+        html.open_in_browser(temp_file_lifetime=None)
+        assert os.path.isfile(html._temp_file)
+        time.sleep(1.6)
+        assert os.path.isfile(html._temp_file)
+    finally:
+        webbrowser.open = wb_open
+        try:
+            os.remove(html._temp_file)
+        except Exception:
+            pass
+
+
+def _open_views():
+    return [html_surface.SurfaceView('') for i in range(12)]
+
+
+def _open_one_view():
+    for i in range(12):
+        v = html_surface.SurfaceView('')
+    return v
+
+
+def test_open_view_warning():
```

Comment:
Maybe tell us in a comment what this test tests (the fact that too many views opened gives a warning)